ta4j-core/src/main/java/org/ta4j/core/indicators/CachedIndicator.java

```diff
@@ -97,7 +97,10 @@ public T getValue(int index) {
                 // Result not calculated yet
                 highestResultIndex = index;
                 result = calculate(index);
-                results.set(results.size()-1, result);
+                if(index != series.getEndIndex()) {
+                    // Only cache results before last bar
+                    results.set(results.size()-1, result);
+                }
```

Comment:
I think this does not work because the following is possible:
**A call of the `getValue(int index)` function with**
- cache= {1,2,3}
- `highestResultIndex `= 2
- `index `= 3 (last bar of series)

1. `increaseLengthTo(index, maximumResultCount)`;
    - cache = {1,2,3,null}
2. `if (index > highestResultIndex) `-> true
    - `highestResultIndex `= 3
    - cache =  {1,2,3,null} (stays the same because index == series.getEndIndex())

**Next call of the `getValue(int index)` function with:**
- `index `= 3 (again)
- cache = {1,2,3,null} 
- `highestResultIndex `= 3
1. `increaseLengthTo(index, maximumResultCount)`;
    - cache = {1, 2, 3, null} (nothing to do)
2. `if (index > highestResultIndex) `-> false!
3. `else...`
    - `resultInnderIndex` = 4-1-0 = 3
    - `result` = null
    - `if (result == null)` -> true
        - `results.set(3, result);` **_## -> last index of indicator is in cache!!!_**


#### Maybe it works if we change `if (index > highestResultIndex)` to `if (index >= highestResultIndex)` ?